File: programmers/k진수에서 소수 개수 구하기/Programmers92335_3.java
package com.example;

import java.util.Arrays;

public class Programmers92335_3 {
    // 1 ≤ n ≤ 1,000,000
    // 3 ≤ k ≤ 10
    public int solution(int n, int k) {
        String kDigit = convertToKDigit(n, k);
        String[] digits = kDigit.replace("0", " ").split("\\s+");
        return (int) Arrays.stream(digits)
                .filter(d -> isPrime(Long.parseLong(d)))
                .count();
    }

    private String convertToKDigit(int n, int k) {
        StringBuilder sb = new StringBuilder();
        while (n > 0) {
            sb.insert(0, n % k);
            n /= k;
        }
        return sb.toString();
    }

    private boolean isPrime(long num) {
        if (num < 2) return false;
        int sqrt = (int) Math.sqrt(num);
        for (int n = 2; n <= sqrt; n++) {
            if (num % n == 0) {
                return false;
            }
        }
        return true;
    }
}

File: programmers/k진수에서 소수 개수 구하기/Programmers92335_3.kt
package com.example

import kotlin.math.sqrt

class Programmers92335_3 {
    fun solution(n: Int, k: Int): Int {
        val kDigit = convertToKDigit(n, k)
        val kDigits = kDigit.split(Regex("0+")).filter { it.isNotBlank() }
        return kDigits.filter { isPrime(it.toLong()) }.size
    }

    private fun convertToKDigit(n: Int, k: Int): String {
        var num = n
        val sb = StringBuilder()
        while (num > 0) {
            sb.insert(0, num % k)
            num /= k
        }
        return sb.toString()
    }

    private fun isPrime(num: Long): Boolean {
        if (num < 2) return false
        val sqrt = sqrt(num.toDouble()).toInt()
        for (i in 2..sqrt) if (num % i == 0L) return false
        return true
    }
}
